### PR TITLE
#GetUserInformation Implements the RPC for querying the current logge…

### PR DESCRIPTION
**Pull request by yuanchenxi95**
#GetUserInformation Implements the RPC for querying the current logged-in user information.

### CODE REVIEW

File: src/main/kotlin/com/yuanchenxi95/twig/utils/databaseutils/SelectOneById.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    return idField!!.name
+}
+
+inline fun <reified T : Any> selectOneById(
+    id: String,
+    r2dbcEntityTemplate: R2dbcEntityTemplate,
+    useDefaultID: Boolean = true
+): Mono<T> {
+    val idFieldName = if (useDefaultID) "id" else findIdKeyField<T>()
+    println(idFieldName)
```
Comment (jiang12358) [inline src/main/kotlin/com/yuanchenxi95/twig/utils/databaseutils/SelectOneById.kt:24]: Remove this line?

File: protobuf/api/user.proto [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+option java_multiple_files = true;
+option java_package = "com.yuanchenxi95.twig.protobuf.api";
+option java_outer_classname = "UserProto";
+
+message GetUserInformationResponse {
+    string id = 1;
+    string email = 2;
+    string name = 3;
+    google.protobuf.Timestamp expiration_time = 4;
```
Comment (jiang12358) [inline protobuf/api/user.proto:15]: Just curious, why we need this field?

File: src/test/kotlin/com/yuanchenxi95/twig/utils/databaseutils/SelectOneByIdKtTest.kt [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package com.yuanchenxi95.twig.utils.databaseutils
+
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.Test
+import org.springframework.data.annotation.Id
+
+class SelectOneByIdKtTest {
```
Comment (jiang12358) [inline src/test/kotlin/com/yuanchenxi95/twig/utils/databaseutils/SelectOneByIdKtTest.kt:7]: Maybe also add a simple test case for selectOneById function?

**Review by jiang12358 — CHANGES_REQUESTED** [review]

File: protobuf/api/user.proto (hunk 2 above)
Comment (yuanchenxi95) [reply protobuf/api/user.proto:15]: For frontend to set the expiration time of the current session.

**Review by jiang12358 — APPROVED** [review]

**Pull request merged**